lib/fetch-playable-url.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const Promise = require('bluebird');
+const request = require('request');
+
+const BASE_URL = 'http://player.ooyala.com/sas/player_api/v1/authorization/embed_code';
+const QUERY = 'device=roku&domain=www.ooyala.com&supportedFormats=m3u8';
+
+// args.apiKey
+// args.embedCode
+//
+// If the playable URL is not defined, this will reject with:
+//   error.code === 'STREAM_UNDEFINED'
+function fetchPlayableUrl(args) {
+	args = args || {};
+	const apiKey = args.apiKey;
+	const embedCode = args.embedCode;
+
+	if (!apiKey) {
+		throw new Error('args.apiKey is required');
+	}
+	if (!embedCode) {
+		throw new Error('args.embedCode is required');
+	}
+
+	const params = {
+		method: 'GET',
+		url: `${BASE_URL}/${apiKey}/${embedCode}?${QUERY}`
+	};
+
+	return fetchPlayableUrl.makeRequest(params).then(data => {
+		const stream = (data.authorization_data[embedCode].streams || [])[0];
+
+		if (!stream || !stream.url || (stream.url.data || '').length < 1) {
+			const error = new Error(
+				`Stream undefined for playable URL with apiKey ${apiKey} and embedCode ${embedCode}.`
+			);
+			error.code = 'STREAM_UNDEFINED';
+			return Promise.reject(error);
+		}
+
+		const buff = new Buffer(stream.url.data, 'base64');
+		const deCoded = buff.toString();
+
+		return deCoded.replace(/player\/iphone/, 'player/appletv');
```

Comment:
FYI - this was a risky move when I implemented it.   As far as I can tell it has been tested very little if at all on rook devices.

I've informed Ooyala that this is a fix that should probably be in place upstream at the api level.